### DISCUSSION TITLE
Sonoff 5-24V as a 230V AC power loss detector

### DISCUSSION THREAD

**Discussion by JayJayXPro** [RavenSystem/esp-homekit-devices#1901] [Q&A]
Hi!

I am trying to create an alarm device on the basis of the Sonoff 5-24V module about the loss of power in the 230V AC network (for the ups system)
The Sonoff 5-24V module will be powered directly from the ups (as well as the modem, apple TV, CCTV and many other devices).

And the optocoupler will be connected to the GPIO4 pin on the sonoff 5-24V, acting as a switch with a state depending on the presence of voltage in the network or not.

MEPLHAA for me is completely rocket science, although I am trying to understand it, could someone help me create the correct code?

Assumptions:

**Normal situation:**

1. Voltage - 230V AC
2. Sensor deactivated

**Exceptional situation:**

1. Voltage - 0V AC
2. Sensor activated
3. Sending an alarm to HomeKit about the loss of power (thus I know that the ups have started supporting the server at home and I need to take the necessary steps)

Thank you in advance for your interest, maybe someone will also need such a device

**Comment by paulzhdanov** [comment]
You can use the switch which listen to GPIO4 to switch: {"g":4},{"g":4,"t":0}. Just insert the code below to your a {} section

    "t":1,
    "0":{"r":[{"g":5,"v":0}]},
    "1":{"r":[{"g":5,"v":1}]},
    "b":[{"g":0},{"g":4},{"g":4,"t":0}]

**Reply by JayJayXPro** [reply]
I did it!
I used your proposal and the sample code for the closure sensor.

It turned out like this:

{
   "c": {"l": 13, "b": [{"g": 0, "t": 5}]},
   "a": [{
     "t": 5,
     "i": 0,
     "0": {"r": [{"g": 16, "v": 0}]},
     "1": {"r": [{"g": 16, "v": 1}]},
     "b": [{"g": 0}, {"g": 14}, {"g": 14, "t": 0}]
   }]
}

Works as expected.
Thanks a lot for your help!

**Comment by komarofleo** [comment]
Dear JayJayXPro
Can you pls provide the diagram or picture of connection of Sonoff module, thanx!
